specification/communication/data-plane/Microsoft.CommunicationServicesCommon/preview/2020-11-19-preview1/common.json

```diff
@@ -0,0 +1,114 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Communication",
+    "description": "Communication Services API common types.",
+    "version": "2020-11-19-preview1"
+  },
+  "paths": {},
+  "definitions": {
+    "CommunicationError": {
+      "description": "The Communication Services error exception.",
+      "required": [
+        "error"
+      ],
+      "properties": {
+        "error": {
+          "description": "The Communication Services error.",
+          "$ref": "#/definitions/Error"
+        }
+      }
+    },
+    "Error": {
```

Comment:
I wonder how is this going to be exposed in the sdks for instance, would it be simply 'Error' ? If so, we probably should find a more explicit type (Error is too generic) . I was running the latest locally and we got warnings :

Severity Code Description Project File Line Suppression State
Error AZC0012 Single word class names are too generic and have high chance of collision with BCL types or types from other libraries Azure.Communication.Chat

Reply:
Good point. I had followed keyvault's example for the model names. https://github.com/Azure/azure-rest-api-specs/blob/master/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.1/common.json

Other services have a generated model type called `Error` but don't expose it in their API. Instead they use to construct and throw an exception.

Reply:
Maybe we can  just merge this, reference it from the Chat swagger and see what we get on the sdk ?
We can follow up with a better name after ..

Reply:
I've renamed to `CommunicationErrorResponse` and `CommunicationError` to be on the safe side.